packages/contract-event-stream/__tests__/node_env.js

```diff
@@ -0,0 +1,162 @@
+    await contractInstance.methods.set(5).send({ from: accounts[0] });
+    const newVal = await contractInstance.methods.get().call();
+
+    expect(oldVal).toBe("0");
+    expect(newVal).toBe("5");
+  });
+
+  test("createContractEvent$ function exists", () => {
+    expect(createContractEvent$).toBeDefined();
+  });
+
+  test("createContractEvent$ throws errors when required options fields not found", () => {
+    const { _address: address } = contractInstance;
+    const { abi } = artifact;
+
+    expect(() => createContractEvent$()).toThrow();
+
+    expect(() => createContractEvent$({ abi, address })).toThrow(
+      new Error("The options object with web3 is required"),
+    );
+
+    expect(() => createContractEvent$({ web3, address })).toThrow(
+      new Error("The options object with contract abi is required"),
+    );
+
+    expect(() => createContractEvent$({ web3, abi })).toThrow(
+      new Error("The options object with contract address is required"),
+    );
+
+    const web3Http = new Web3("http://127.0.0.1:9545"); // HttpProvider
+    expect(() =>
+      createContractEvent$({ web3: web3Http, abi, address }),
+    ).toThrow(
+      new Error("Must provide newBlock$ when using http provider with web3"),
+    );
+  });
+
+  test("fromPolling can track events emitted by send method", async done => {
+    const { observable: newBlock$, subscription } = createNewBlock$({
+      web3,
+      pollingInterval: 200,
+    });
+
+    const { _address: address } = contractInstance;
+    const { abi } = artifact;
+    const event$ = createContractEvent$({ web3, abi, address, newBlock$ });
+
+    // tap observable to make sure it emitted a "0" and then a "5"
+    event$
+      .pipe(
+        take(2),
+        toArray(),
+        tap(vals =>
+          vals.forEach(val => {
+            expect(val).toMatchSnapshot(makePropertyMatcher());
+          }),
+        ),
+        finalize(() => {
+          expect.assertions(2);
+          subscription.unsubscribe();
+          done();
+        }),
+      )
+      .subscribe();
+
+    await contractInstance.methods.set(0).send({ from: accounts[0] });
+    await contractInstance.methods.set(5).send({ from: accounts[0] });
+  });
+
+  test("fromSubscribe can track events emitted by send method", async done => {
+    // Change constructor so we can test Websocket route
+    // Note that ganache provider.constructor.name is Provider
+    function WebsocketProvider() {}
+    const web3Ws = new Web3(provider);
+    web3Ws.currentProvider.constructor = WebsocketProvider;
+
+    const { _address: address } = contractInstance;
+    const { abi } = artifact;
+    const event$ = createContractEvent$({ web3: web3Ws, abi, address });
+
+    // tap observable to make sure it emitted a "0" and then a "5"
+    event$
+      .pipe(
+        take(2),
+        toArray(),
+        tap(vals => {
+          vals.forEach(val => {
+            expect(val).toMatchSnapshot(makePropertyMatcher());
```

Comment:
Naming isn't so clear here. You have `makePropertyMatcher()`, but I assume what you want to say is `isEventObject`?